lib/cve.py

```diff
@@ -51,82 +50,127 @@ def read_json(file):
 def get_component_names(data_list):
     component_list = []
     for component_info in data_list:
+        log(f"component_info: {component_info}")
         for key, value in component_info.items():
             if (key == "name"):
                 component_list.append(value)
     return component_list
 
 
 def get_component_detail(data_list, component):
-    found = 0
+    log(f"looking for component detail: {component}")
     for component_info in data_list:
-        for key, value in component_info.items():
-            if (key == "name") and (value == component):
-                found = 1
-            if (key == "source"):
-                return(value["git"]["url"],value["git"]["revision"])
+        log(f"component_info: {components_info}")
+        if component == component_info["name"]:
+            return component_info["source"]["git"]["url"], component_info["source"]["git"]["revision"]
     return([])
 
 
+def get_snapshot_data(namespace, snapshot):
+    cmd = ["kubectl", "get", "snapshot", snapshot, "-n", namespace, "-ojson"]
+    try:
+        cmd_str = " ".join(cmd)
+        log(f"Running {cmd_str}")
+        result = subprocess.run(cmd, check=True, capture_output=True, text=True)
+    except subprocess.CalledProcessError:
+        log(f"Command {cmd_str} failed, check exception for details")
+        raise
+    except Exception as exc:
+        log("Unknown error occurred")
+        raise RuntimeError from exc
+
+    log(result.stdout)
+    return json.loads(result.stdout)
+
+
+def log(message):
+    print(message, file=sys.stderr)
+
+
+def get_snapshot_name(data_release):
+    return data_release['spec']['snapshot']
+
+
+def get_snapshot_namespace(data_release):
+    return data_release['metadata']['namespace']
```

Comment:
This method should check for the existence of `namespace` in the structure and provide a clear error message if it does not exist.

Reply:
fixed